File: backend/app/models/basketball_club.py
from sqlalchemy import Column, Integer, String, DateTime, Text, JSON, Index
from sqlalchemy.sql import func
from app.db.session import Base

class BasketballClub(Base):
    __tablename__ = "basketball_clubs"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, index=True, nullable=False)
    city = Column(String, index=True)
    country = Column(String, index=True)
    league = Column(String, index=True) # NBA, EuroLeague, etc.
    conference = Column(String) # Eastern, Western
    founded_year = Column(Integer)
    arena = Column(String)
    capacity = Column(Integer)
    head_coach = Column(String)
    nickname = Column(String)
    image_url = Column(String)
    website = Column(String)
    description = Column(Text)
    
    # Rankings & Stats
    ranking = Column(Integer, index=True) # Power Ranking
    titles = Column(Integer, default=0) # e.g., NBA Championships
    playoff_appearances = Column(Integer, default=0)
    market_value = Column(String)
    current_season_record = Column(String) # e.g., "45-20"
    
    # Key Personnel
    star_player = Column(String)
    owner = Column(String)
    general_manager = Column(String)
    
    # Detailed Honors
    # Example: {"NBA Titles": 17, "Conference Titles": 32, "Division Titles": 25}
    honors_json = Column(JSON)

    last_updated = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    __table_args__ = (
        Index("ix_basketball_clubs_name_lower", func.lower(name)),
    )

File: frontend/lib/models/basketball_club.dart
class BasketballClub {
  final int id;
  final String name;
  final String? city;
  final String? country;
  final String? league;
  final String? conference;
  final int? foundedYear;
  final String? arena;
  final int? capacity;
  final String? headCoach;
  final String? nickname;
  final String? imageUrl;
  final String? website;
  final String? description;
  final int? ranking;
  final String category;

  // Stats & Personnel
  final int titles;
  final int playoffAppearances;
  final String? marketValue;
  final String? currentSeasonRecord;
  final String? starPlayer;
  final String? owner;
  final String? generalManager;

  // Detailed Honors
  final Map<String, int>? honors;

  BasketballClub({
    required this.id,
    required this.name,
    this.city,
    this.country,
    this.league,
    this.conference,
    this.foundedYear,
    this.arena,
    this.capacity,
    this.headCoach,
    this.nickname,
    this.imageUrl,
    this.website,
    this.description,
    this.ranking,
    required this.category,
    this.titles = 0,
    this.playoffAppearances = 0,
    this.marketValue,
    this.currentSeasonRecord,
    this.starPlayer,
    this.owner,
    this.generalManager,
    this.honors,
  });

  factory BasketballClub.fromJson(Map<String, dynamic> json) {
    return BasketballClub(
      id: json['id'],
      name: json['name'],
      city: json['city'],
      country: json['country'],
      league: json['league'],
      conference: json['conference'],
      foundedYear: json['founded_year'],
      arena: json['arena'],
      capacity: json['capacity'],
      headCoach: json['head_coach'],
      nickname: json['nickname'],
      imageUrl: json['image_url'],
      website: json['website'],
      description: json['description'],
      ranking: json['ranking'],
      category: json['category'] ?? 'men',
      titles: json['titles'] ?? 0,
      playoffAppearances: json['playoff_appearances'] ?? 0,
      marketValue: json['market_value'],
      currentSeasonRecord: json['current_season_record'],
      starPlayer: json['star_player'],
      owner: json['owner'],
      generalManager: json['general_manager'],
      honors: json['honors_json'] != null
          ? Map<String, int>.from(json['honors_json'])
          : null,
    );
  }
}

class BasketballClubListResponse {
  final List<BasketballClub> items;
  final int total;
  final int page;
  final int size;

  BasketballClubListResponse({
    required this.items,
    required this.total,
    required this.page,
    required this.size,
  });

  factory BasketballClubListResponse.fromJson(Map<String, dynamic> json) {
    return BasketballClubListResponse(
      items: (json['items'] as List)
          .map((i) => BasketballClub.fromJson(i))
          .toList(),
      total: json['total'],
      page: json['page'],
      size: json['size'],
    );
  }
}
